ev3dev/core.py

```diff
@@ -3046,100 +3135,49 @@ def buttons_pressed(self):
         return pressed
 
 
-class RemoteControl(ButtonBase):
+class RemoteControl(ButtonBase, InfraredSensor):
     """
     EV3 Remote Controller
     """
 
-    _BUTTON_VALUES = {
-            0: [],
-            1: ['red_up'],
-            2: ['red_down'],
-            3: ['blue_up'],
-            4: ['blue_down'],
-            5: ['red_up', 'blue_up'],
-            6: ['red_up', 'blue_down'],
-            7: ['red_down', 'blue_up'],
-            8: ['red_down', 'blue_down'],
-            9: ['beacon'],
-            10: ['red_up', 'red_down'],
-            11: ['blue_up', 'blue_down']
-            }
-
-    #: Handles ``Red Up`` events.
-    on_red_up = None
-
-    #: Handles ``Red Down`` events.
-    on_red_down = None
-
-    #: Handles ``Blue Up`` events.
-    on_blue_up = None
-
-    #: Handles ``Blue Down`` events.
-    on_blue_down = None
-
-    #: Handles ``Beacon`` events.
-    on_beacon = None
+    def __init__(self, sensor=None, channel=1):
```

Comment:
Is this the order we want it in? Or should `channel` be first? I'd imagine that providing a custom sensor will be quite rare. Not that it _really_ matters as you can reference it by name anyway.

Reply:
It was that way before so we should probably leave it as-is

Reply:
The PR makes the class a subclass of `InfraredSensor`. Why do we need the `sensor` parameter (and `_sensor` field) here at all?

Reply:
for backwards compatibility

Reply:
I mean, the `BeaconSeeker` class here already is `InfraredSensor` (since it inherits from it). So why store another instance of `InfraredSensor` in `_sensor`?

Reply:
Because our current API allows the user to pass in a InfraredSensor object to the BeaconSeeker and use that as the _sensor.  If we are taking the stance that the new API is not backwards compatible with the old API then we can certainly clean all of this up a good bit. My understanding was that we wanted to keep the old API around for backward compatibility.

Reply:
Then we should not derive from `InfraredSensor`.

Reply:
I think we should simply be deliberate about it. If there's a good reason to make a breaking change, I would rather do it now for the first Stretch version than later on.

Reply:
Given how much code/API we can clean up if we are not backwards compatible I think it would be worth it.  ok just so we are all on the same page, we are saying drop the `RemoteSensor` and `BeaconSeeker` classes and put all functionality in the `InfraredSensor` class yes?

Reply:
I think that seems reasonable; how does this map to EV3-G?

Reply:
EV3-G has the equivalent of our `InfraredSensor`, `RemoteSensor`, and `BeaconSeeker` all in a single block.  So we put all of the logic in `InfraredSensor`, delete `RemoteSensor` and `BeaconSeeker` and then we are 1:1 with the EV3-G block.

Reply:
:+1: I support this

Reply:
python zen: "flat is better than nested"

having a single InfraredSensor class sounds like a good call.

Reply:
We should start keeping track of all breaking changes; do you guys want to do that in the changelog file as we go or add a "changelog" section to the README? Or maybe create an issue thread to post into? I'm up for whatever's easiest for you guys.

Reply:
We can start a draft of new release [here](https://github.com/rhempel/ev3dev-lang-python/releases). It seems README could grow too long too fast if we start putting changes there. Another idea is we could add another section in the documentation for this.

Reply:
Interesting idea! I just created a release where we can try out tracking this; as we merge PRs we should make sure to update it. If this doesn't go well we can try out a more conventional file approach. https://github.com/rhempel/ev3dev-lang-python/releases/tag/untagged-f43144149d22a7d3cc09

Note that the link might  break once we add a version tag.